lib/pbench/server/api/resources/query_controllers.py

```diff
@@ -0,0 +1,273 @@
+from flask import request
+from flask_restful import Resource, abort
+import requests
+
+from dateutil import parser, rrule
+
+
+class QueryControllers(Resource):
+    """
+    Abstracted Pbench API to get date-bounded controller data.
+    """
+
+    def __init__(self, app, api):
+        self.app = app
+        self.api = api
+
+    @staticmethod
+    def _genMonthUriRange(prefix, index, start, end):
+        """
+        _genMonthUriRange Construct a comma-separated list of index names
+        qualified by year and month suitable for use in the Elasticsearch
+        /_search query URI.
+
+        The month is incremented by 1 from "start" to "end"; for example,
+
+        _genMonthUriRange('drb.', 'v4.run.', '2020-08', '2020-10') will result
+        in
+
+            'drb.v4.run.2020-08,drb.v4.run.2020-09,drb.v4.run.2020-10,'
+
+        This code is adapted from Javascript in the Pbench dashboard's
+        'moment_constants.js' getAllMonthsWithinRange function.
+
+        Args:
+            prefix ([string]): The Pbench server index prefix
+            index ([string]): The desired monthly index root
+            start ([datetime.datetime]): The start time
+            end ([datetime.datetime]): The end time
+
+        Returns:
+            [string]: A comma-separated list of month-qualified index names
+        """
+        monthResults = list()
+        queryString = ""
+        for m in rrule.rrule(rrule.MONTHLY, dtstart=start, until=end):
+            monthResults.append(m.strftime("%Y-%m"))
+        for monthValue in monthResults:
+            if index == "v4.result-data.":
```

Comment:
This is kinda intimate knowledge of the index name pattern.  Could we pass an argument to direct the addition of the `-*` suffix instead?

Reply:
I really need to work out a better model for handling index versioning; this check comes directly from the Javascript. It's not even immediately relevant since it's not used (yet) for result data. This really should key off the index_patterns[root_name]['template_pat'] data, dynamically... but we don't have that until I work out a way to process and cache the templates.

I left this more as a template and marker than anything else; but maybe I should at least add a TODO here so it doesn't get lost.